R/utils.R

```diff
@@ -538,9 +538,7 @@ installExprFunction <- function(expr, name, eval.env = parent.frame(2),
 #' @export
 quoToFunction <- function(q, label, ..stacktraceon = FALSE) {
   q <- as_quosure(q)
-  # Use new_function() instead of as_function(), because as_function() adds an
-  # extra parent environment. (This may not actually be a problem, though.)
-  func <- new_function(NULL, get_expr(q), get_env(q))
+  func <- as_function(q)
```

Comment:
Note that this is causing a test failure because the old code creates a function with the signature `function()`, but the new code creates a function with the signature `function(...)`. I don't think that the added `...` currently cause any problems right now (other than the test failure), but in order to keep the code tighter, I think we should remove them. I can make the change.